src/sql/SqlResult.ts

```diff
@@ -330,8 +307,6 @@ export class SqlResultImpl implements SqlResult {
             this.updateCount = updateCount;
             this.closed = true;
         }
-        this.responseReceived = true;
```

Comment:
There are some refactorings that can be done here. If you want, you can do it later. It is not required for correctness but for example, we might remove the code that assumes the user can close the result before the response arrives.